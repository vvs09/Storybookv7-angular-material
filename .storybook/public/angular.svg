<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
