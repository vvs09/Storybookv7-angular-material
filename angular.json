{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}