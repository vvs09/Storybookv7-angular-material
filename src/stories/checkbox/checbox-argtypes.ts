export const checkboxArgTypes = {
  color: {
    control: 'radio',
    options: ['accent', 'primary', 'warn'],
    description: 'Theme color palette for the component.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'undefined' },
      type: {
        summary: 'string  | undefined',
      },
    },
  },
  disabled: {
    control: 'boolean',
    description:
      'Whether the checkbox is disabled. This fully overrides the implementation provided by mixinDisabled, but the mixin is still required because mixinTabIndex requires it.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
  ariaDescribedby: {
    control: 'text',
    description:
      'The aria-describedby attribute is read after the elements label and field type.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  ariaLabel: {
    control: 'text',
    description: 'ARIA label for the content of the chip.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string  | undefined',
      },
    },
  },
  disableRipple: {
    control: 'boolean',
    description: 'Whether ripples are disabled.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
  ariaLabelledby: {
    control: 'text',
    description:
      'Users can specify the aria-labelledby attribute which will be forwarded to the input element',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  id: {
    control: 'text',
    description:
      'A unique id for the checkbox input. If none is supplied, it will be auto-generated.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  checked: {
    control: 'boolean',
    description: 'Whether the checkbox is checked.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'boolean',
      },
    },
  },
  indeterminate: {
    control: 'boolean',
    description:
      'Whether the checkbox is indeterminate. This is also known as "mixed" mode and can be used to represent a checkbox with three states, e.g. a checkbox that represents a nested list of checkable items. Note that whenever checkbox is manually clicked, indeterminate is immediately set to false.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'boolean',
      },
    },
  },
  labelPosition: {
    control: 'radio',
    options: ['before', 'after'],
    description:
      'Whether the label should appear after or before the checkbox. Defaults to after',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'after' },
      type: {
        summary: 'before | after',
      },
    },
  },
  name: {
    control: 'text',
    description: 'Name value will be applied to the input element if present',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  required: {
    control: 'boolean',
    description: 'Whether the checkbox is required.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'boolean',
      },
    },
  },
  change: {
    control: 'object',
    description: 'Event emitted when the checkboxs checked value changes.',
    table: {
      category: 'OUTPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'EventEmitter<MatCheckboxChange',
      },
    },
  },
  indeterminateChange: {
    control: 'object',
    description:
      'Event emitted when the checkboxs indeterminate value changes.',
    table: {
      category: 'OUTPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'EventEmitter<boolean>',
      },
    },
  },
  inputId: {
    control: 'text',
    description: 'Returns the unique id for the visual hidden input.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  value: {
    control: 'text',
    description: 'The value attribute of the native input element.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
};
