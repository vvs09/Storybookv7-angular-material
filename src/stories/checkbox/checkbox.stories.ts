import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatCheckboxModule, MatCheckbox } from '@angular/material/checkbox';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { checkboxArgTypes } from './checbox-argtypes';

const meta: Meta<MatCheckbox> = {
  title: 'Components/Checkbox',
  component: MatCheckbox,
  tags: ['autodocs'],
  argTypes: {
    ...checkboxArgTypes,
  },
  decorators: [
    moduleMetadata({
      imports: [ReactiveFormsModule, FormsModule, MatCheckboxModule],
    }),
  ],
  render: (args) => ({
    props: args,
    template: ` <mat-checkbox ${argsToTemplate(args)}>Checkbox</mat-checkbox>`,
  }),
};

export default meta;
type Story = StoryObj<MatCheckbox>;

export const Default: Story = {
  args: {
    color: 'accent',
    disabled: false,
    ariaDescribedby: 'checkbox',
    ariaLabel: 'checkbox',
    ariaLabelledby: 'checkbox',
    disableRipple: false,
    id: '1',
    checked: false,
    indeterminate: false,
    labelPosition: 'after',
    name: 'label checkbox',
    required: false,
    inputId: '1',
    value: 'checkbox1',
  },
};

export const Primary: Story = {
  args: {
    ...Default.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...Default.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...Default.args,
    disabled: true,
  },
};

export const Checked: Story = {
  args: {
    ...Default.args,
    checked: true,
  },
};

export const Indeterminate: Story = {
  args: {
    ...Default.args,
    indeterminate: true,
  },
};

export const LabelPosition: Story = {
  args: {
    ...Default.args,
    labelPosition: 'before',
  },
};

export const Required: Story = {
  args: {
    ...Default.args,
    required: true,
  },
};
