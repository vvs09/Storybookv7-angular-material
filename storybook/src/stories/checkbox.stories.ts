import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatCheckboxModule, MatCheckbox } from '@angular/material/checkbox';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

const meta: Meta<MatCheckbox> = {
  title: 'Components/Checkbox',
  component: MatCheckbox,
  tags: ['autodocs'],
  argTypes: {
    color: {
      control: 'radio',
      options: ['accent', 'primary', 'warn'],
      description: 'Theme color palette for the component.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    disabled: {
      control: 'boolean',
      description:
        'Whether the checkbox is disabled. This fully overrides the implementation provided by mixinDisabled, but the mixin is still required because mixinTabIndex requires it.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    ariaDescribedby: {
      control: 'text',
      description:
        'The aria-describedby attribute is read after the elements label and field type.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    ariaLabel: {
      control: 'text',
      description: 'ARIA label for the content of the chip.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    disableRipple: {
      control: 'boolean',
      description: 'Whether ripples are disabled.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    ariaLabelledby: {
      control: 'text',
      description:
        'Users can specify the aria-labelledby attribute which will be forwarded to the input element',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    id: {
      control: 'text',
      description:
        'A unique id for the checkbox input. If none is supplied, it will be auto-generated.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    checked: {
      control: 'boolean',
      description: 'Whether the checkbox is checked.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    indeterminate: {
      control: 'boolean',
      description:
        'Whether the checkbox is indeterminate. This is also known as "mixed" mode and can be used to represent a checkbox with three states, e.g. a checkbox that represents a nested list of checkable items. Note that whenever checkbox is manually clicked, indeterminate is immediately set to false.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    labelPosition: {
      control: 'radio',
      options: ['before', 'after'],
      description:
        'Whether the label should appear after or before the checkbox. Defaults to after',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    name: {
      control: 'text',
      description: 'Name value will be applied to the input element if present',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    required: {
      control: 'boolean',
      description: 'Whether the checkbox is required.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    change: {
      control: 'object',
      description: 'Event emitted when the checkboxs checked value changes.',
      table: {
        category: 'OUTPUTS',
        defaultValue: { summary: '-' },
      },
    },
    indeterminateChange: {
      control: 'object',
      description:
        'Event emitted when the checkboxs indeterminate value changes.',
      table: {
        category: 'OUTPUTS',
        defaultValue: { summary: '-' },
      },
    },
    inputId: {
      control: 'text',
      description: 'Returns the unique id for the visual hidden input.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    ripple: {
      control: 'object',
      description: 'Reference to the ripple instance of the checkbox.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    value: {
      control: 'text',
      description: 'The value attribute of the native input element.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
  },
  decorators: [
    moduleMetadata({
      imports: [ReactiveFormsModule, FormsModule, MatCheckboxModule],
    }),
  ],
  render: (args: MatCheckbox) => ({
    props: args,
    template: ` <mat-checkbox ${argsToTemplate(args)}>Checkbox</mat-checkbox>`,
  }),
};

export default meta;
type Story = StoryObj<MatCheckbox>;

export const Default: Story = {
  args: {
    color: 'accent',
    disabled: false,
    ariaDescribedby: 'checkbox',
    ariaLabel: 'checkbox',
    ariaLabelledby: 'checkbox',
    disableRipple: false,
    id: '1',
    checked: false,
    indeterminate: false,
    labelPosition: 'after',
    name: 'label checkbox',
    required: false,
    inputId: '1',
    value: 'checkbox1',
  },
};

export const Primary: Story = {
  args: {
    ...Default.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...Default.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...Default.args,
    disabled: true,
  },
};

export const Checked: Story = {
  args: {
    ...Default.args,
    checked: true,
  },
};

export const Indeterminate: Story = {
  args: {
    ...Default.args,
    indeterminate: true,
  },
};

export const LabelPosition: Story = {
  args: {
    ...Default.args,
    labelPosition: 'before',
  },
};

export const Required: Story = {
  args: {
    ...Default.args,
    required: true,
  },
};
