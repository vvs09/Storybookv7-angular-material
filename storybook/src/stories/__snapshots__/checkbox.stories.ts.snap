// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Checkbox Checked smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-accent mat-mdc-checkbox-checked"
                ng-reflect-color="accent"
                ng-reflect-disabled="false"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="true"
                ng-reflect-indeterminate="false"
                ng-reflect-label-position="after"
                ng-reflect-name="label checkbox"
                ng-reflect-required="false"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control mdc-checkbox--selected"
               id="1-input"
               tabindex="0"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;

exports[`Components/Checkbox Default smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-accent"
                ng-reflect-color="accent"
                ng-reflect-disabled="false"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="false"
                ng-reflect-indeterminate="false"
                ng-reflect-label-position="after"
                ng-reflect-name="label checkbox"
                ng-reflect-required="false"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control"
               id="1-input"
               tabindex="0"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;

exports[`Components/Checkbox Disabled smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-accent mdc-checkbox--disabled mat-mdc-checkbox-disabled"
                ng-reflect-color="accent"
                ng-reflect-disabled="true"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="false"
                ng-reflect-indeterminate="false"
                ng-reflect-label-position="after"
                ng-reflect-name="label checkbox"
                ng-reflect-required="false"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control"
               disabled
               id="1-input"
               tabindex="-1"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="true"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;

exports[`Components/Checkbox Indeterminate smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-accent"
                ng-reflect-color="accent"
                ng-reflect-disabled="false"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="false"
                ng-reflect-indeterminate="true"
                ng-reflect-label-position="after"
                ng-reflect-name="label checkbox"
                ng-reflect-required="false"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control"
               id="1-input"
               tabindex="0"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;

exports[`Components/Checkbox LabelPosition smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-accent"
                ng-reflect-color="accent"
                ng-reflect-disabled="false"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="false"
                ng-reflect-indeterminate="false"
                ng-reflect-label-position="before"
                ng-reflect-name="label checkbox"
                ng-reflect-required="false"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field mdc-form-field--align-end">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control"
               id="1-input"
               tabindex="0"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;

exports[`Components/Checkbox Primary smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-primary"
                ng-reflect-color="primary"
                ng-reflect-disabled="false"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="false"
                ng-reflect-indeterminate="false"
                ng-reflect-label-position="after"
                ng-reflect-name="label checkbox"
                ng-reflect-required="false"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control"
               id="1-input"
               tabindex="0"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;

exports[`Components/Checkbox Required smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-accent"
                ng-reflect-color="accent"
                ng-reflect-disabled="false"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="false"
                ng-reflect-indeterminate="false"
                ng-reflect-label-position="after"
                ng-reflect-name="label checkbox"
                ng-reflect-required="true"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control"
               id="1-input"
               required
               tabindex="0"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;

exports[`Components/Checkbox Warn smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-checkbox class="mat-mdc-checkbox mat-warn"
                ng-reflect-color="warn"
                ng-reflect-disabled="false"
                ng-reflect-disable-ripple="false"
                ng-reflect-id="1"
                ng-reflect-checked="false"
                ng-reflect-indeterminate="false"
                ng-reflect-label-position="after"
                ng-reflect-name="label checkbox"
                ng-reflect-required="false"
                ng-reflect-value="checkbox1"
                id="1"
  >
    <div class="mdc-form-field">
      <div class="mdc-checkbox">
        <div class="mat-mdc-checkbox-touch-target">
        </div>
        <input type="checkbox"
               class="mdc-checkbox__native-control"
               id="1-input"
               tabindex="0"
               name="label checkbox"
               value="checkbox1"
        >
        <div class="mdc-checkbox__ripple">
        </div>
        <div class="mdc-checkbox__background">
          <svg focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               class="mdc-checkbox__checkmark"
          >
            <path fill="none"
                  d="M1.73,12.91 8.1,19.28 22.79,4.59"
                  class="mdc-checkbox__checkmark-path"
            >
            </path>
          </svg>
          <div class="mdc-checkbox__mixedmark">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-mdc-checkbox-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
        </div>
      </div>
      <label for="1-input">
        Checkbox
      </label>
    </div>
  </mat-checkbox>
</storybook-root>
`;
